apps/shipments/permissions.py

```diff
@@ -92,3 +98,15 @@ def has_permission(self, request, view):
             return True
 
         return super(IsAuthenticatedOrDevice, self).has_permission(request, view)
+
+
+class DeviceShipmentHistoryPermission(permissions.BasePermission):
```

Comment:
I think we can just use the default Shipment permissions check, since device_id is a field on the Shipment, not the device.